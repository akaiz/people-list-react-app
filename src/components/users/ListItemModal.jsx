import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Avatar from 'react-avatar';
import { Modal } from 'react-bootstrap';
import _ from 'lodash';

function ListItemModal({
  data, showModal, toggleModal, deleteUser
}) {
  const [show, setShow] = useState(false);
  useEffect(() => {
    if (showModal !== show) {
      setShow(true);
    }
  }, [showModal]);

  const handleClose = () => { setShow(false); toggleModal(false); };
  const assistant = data.e80927bf538ab0f90854b97d28a677a12639d62a;
  const groups = data.cbc8ad54003239431c1cde5c2d79378b8c4c5211;
  const location = data['783dafde54ea51ce69139af515955b60df27219a'];
  const fields = {
    Email: data.email ? data.email[0].value : '',
    Organization: data.org_name ? data.org_name : '',
    Assistant: assistant || '',
    Groups: groups || '',
    Location: location || '',
  };
  const phoneNumber = _.find(data.phone, { primary: true });
  const onDeleteUser = () => { deleteUser(data); setShow(false); toggleModal(false); };
  return (
    <>
      <Modal
        show={show}
        onHide={handleClose}
        dialogClassName="modal-80w"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header closeButton>
          <h5 className="modal-header-content">Person Information</h5>
        </Modal.Header>
        <Modal.Body>
          <div>
            <div className="text-center  modal-user-row-1">
              <Avatar src={data.user_photo} round="50px" name={data.name} color="#d1dff0" fgColor="#357edd" />
              <h4 className="modal-user-name">{data.name}</h4>
              <h5 className="modal-user-phone">{phoneNumber ? phoneNumber.value : ''}</h5>
            </div>
            <div className="modal-user-separator"></div>
            <div className="modal-user-row-2">
              {
                Object.keys(fields).map((key) => (
                  <div className="row py-2" key={key}>
                    <h6 className="col-xs-3 col-sm-6 col-md-4 modal-user-field">{key}</h6>
                    <h6 className="col-xs-7 col-sm-6  col-md-8 modal-user-value">{fields[key]}</h6>
                  </div>
                ))
              }
            </div>

          </div>
        </Modal.Body>
        <Modal.Footer>
          <button className="btn modal-delete-button mr-auto" onClick={onDeleteUser}>
            Delete
          </button>
          <button className="btn modal-back-button col-sm-3" onClick={handleClose}>
            Back
          </button>
        </Modal.Footer>
      </Modal>
    </>
  );
}

ListItemModal.propTypes = {
  data: PropTypes.object.isRequired,
  showModal: PropTypes.bool.isRequired,
  toggleModal: PropTypes.func.isRequired,
  deleteUser: PropTypes.func.isRequired,
};
export default ListItemModal;
