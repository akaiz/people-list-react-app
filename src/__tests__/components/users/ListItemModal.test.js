import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Avatar from 'react-avatar';

import ListItemModal from '../../../components/users/ListItemModal';

import { apiResponse } from '../../TestHelper';

configure({ adapter: new Adapter() });

const setUp = (props = {}) => {
  const component = mount(
    <ListItemModal {...props} />);
  return component;
};

describe('<ListItemModal> Tests', () => {
  let wrapper;
  const user = apiResponse(1).data[0];
  const props = {
    data: user, showModal: true, toggleModal: jest.fn(() => {}), deleteUser: jest.fn(() => {})
  };
  beforeEach(() => {
    wrapper = setUp(props);
  });

  describe('Checking Props', () => {
    it('Should receive the sent props', () => {
      expect(wrapper.find(ListItemModal).props()).toStrictEqual(props);
    });
  });

  describe('Check rendered details', () => {
    it('Renders correctly', () => {
      const tree = renderer
        .create(<ListItemModal {...props} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('Should render user name ', () => {
      expect(wrapper.find('.modal-user-name').text()).toBe(props.data.name);
    });

    it('Should render user phone ', () => {
      expect(wrapper.find('.modal-user-phone').text().trim()).toMatch(user.phone[0].value);
    });

    it('Should render user photo ', () => {
      expect(wrapper.find(Avatar).length).toBe(1);
    });
  });
});
