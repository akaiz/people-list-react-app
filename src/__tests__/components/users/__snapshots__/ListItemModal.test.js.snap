// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemModal> Tests Check rendered details Renders correctly 1`] = `null`;

exports[`<ListItemModal> Tests Check rendered details renders correctly 1`] = `null`;
